### PR TITLE
hv: fix a few bugs for msix.c

### PR DESCRIPTION
**Pull request by ZideChen0**
Tracked-On: #1568
Signed-off-by: Zide Chen <zide.chen@intel.com>

### CODE REVIEW

File: hypervisor/dm/vpci/msix.c [hunk 1]
```diff
@@ -217,10 +217,22 @@ static void vmsix_table_rw(struct pci_vdev *vdev, struct mmio_request *mmio, uin
...
+		/*
+		 * Writing different value to Message Data/Addr?
+		 * PCI Spec: Software is permitted to fill in MSI-X Table entry DWORD fields individually
+		 * with DWORD writes, or software in certain cases is permitted to fill in appropriate pairs
+		 * of DWORDs with a single QWORD write
+		 */
+		if (entry_offset < offsetof(struct msix_table_entry, data)) {
+			uint64_t qword_mask = ~0U;
+			if (mmio->size == 4U) {
+				qword_mask = (entry_offset == 0U) ? 0x00000000FFFFFFFFU : 0xFFFFFFFF00000000U;
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/msix.c]: This UL, right?

**Comment by HuiHuangShi** [comment]
1851<br>No new violations to the coding guideline detected.

File: hypervisor/dm/vpci/msix.c (hunk 1 above)
Comment (ZideChen0) [reply hypervisor/dm/vpci/msix.c]: Yes, it should be UL. It passed my unit test because "0xFFFFFFFF00000000U" is treated as 64 bits, not 32 bits.

Comment (dongyaozu) [reply hypervisor/dm/vpci/msix.c]: use UL to make it clear, rather than implicit type casting

**Comment by HuiHuangShi** [comment]
1851<br>No new violations to the coding guideline detected.

File: hypervisor/dm/vpci/msix.c (hunk 1 above)
Comment (ZideChen0) [reply hypervisor/dm/vpci/msix.c]: Yes, I have updated the PR.

**Pull request merged**